kernel/tests/v2_checkpoints.rs

```diff
@@ -0,0 +1,236 @@
+// rustfmt::skip
+use std::sync::Arc;
+
+use arrow_array::RecordBatch;
+use delta_kernel::engine::sync::SyncEngine;
+
+use delta_kernel::engine::arrow_data::ArrowEngineData;
+use delta_kernel::{DeltaResult, Table};
+
+mod common;
+use common::{load_test_data, read_scan};
+
+fn read_v2_checkpoint_table(test_name: impl AsRef<str>) -> DeltaResult<Vec<RecordBatch>> {
+    let test_dir = load_test_data("tests/data", test_name.as_ref()).unwrap();
+    let test_path = test_dir.path().join(test_name.as_ref());
+
+    let table = Table::try_from_uri(test_path.to_str().expect("table path to string")).unwrap();
+    let engine = Arc::new(SyncEngine::new());
+    let snapshot = table.snapshot(engine.as_ref(), None)?;
+    let scan = snapshot.into_scan_builder().build()?;
+    let batches = read_scan(&scan, engine)?;
+
+    Ok(batches)
+}
+
+fn test_v2_checkpoints(table_name: &str) -> DeltaResult<()> {
+    let num_of_max_digits = 999.to_string().len();
+    let batches = read_v2_checkpoint_table(table_name)?;
+
+    fn generate_rows(count: usize, width: usize) -> Vec<String> {
+        (0..count)
+            .map(|id| format!("| {:<width$} |", id, width = width))
+            .collect()
+    }
+
+    let mut expected = vec![
+        "+-----+".to_string(),
+        "| id  |".to_string(),
+        "+-----+".to_string(),
+    ];
+
+    // Generate rows of expected table in the same manner as delta-spark creates them
+    expected.extend(vec!["| 0   |".to_string(); 3]);
+    let all_batches = [
+        generate_rows(30, num_of_max_digits),
+        generate_rows(100, num_of_max_digits),
+        generate_rows(100, num_of_max_digits),
+        generate_rows(1000, num_of_max_digits),
+    ];
```

Comment:
This is a bit of a long shot and idk if it's too fancy, but does this work?
```suggestion
    let header = vec![
        "+-----+".to_string(),
        "| id  |".to_string(),
        "+-----+".to_string(),
    ];
    let all_batches = [
        header
        vec!["| 0 |".to_string(); 3],
        generate_rows(30, num_of_max_digits),
        generate_rows(100, num_of_max_digits),
        generate_rows(100, num_of_max_digits),
        generate_rows(1000, num_of_max_digits),
        vec!["+-----+".to_string()]
    ]
    .flat_map(identity);
```